### DISCUSSION TITLE
Validate does not work with drafts enabled

### DISCUSSION THREAD

**Discussion by silveltman** [payloadcms/payload#3576] [Roadmap]
### Link to reproduction

/

### To Reproduce

1. Enable drafts
2. Create a collection with the following field:
```
    {
      name: 'slug',
      label: 'Slug',
      type: 'text',
      unique: true,
      validate: ({ value }) => {
        return 'This is not allowed';
      },
    },
```
3. Click 'save draft' button
4. then click 'publish' button

### Describe the Bug

You will see that the validation error only appears when clicking publish, and not when clicking save draft.
This is a problem in my case, because I'm trying to do the following:

When someone changes a slug, check if a siblingdata checkbox called 'confirm' is true, so that it adds an extra step for changing a slug. 

### Payload Version

2.0.3

### Adapters and Plugins

mongo, vite

**Comment by silveltman** [comment]
Same for 'required'. This an even bigger problems, since it can give all sort of errors on the frontend.

The frontend expect a value to be there, since it is required. When running a preview website, where API requests are caleld with ?draft=true parameter, this will raise problems

**Comment by jmikrut** [comment]
Hey @silveltman — i see what you are running into here. The problem is that drafts can deliberately be saved as _**partially complete**_, and this is intentional in a lot of cases. It is also _**required**_ for `autosave` enabled. 

It's possible that we can expose a new setting so that if you have drafts enabled, but not autosave, we could prevent incomplete drafts from being saved.

This is a feature request though because this is technically functioning as intended. I think we can look into exposing a new option here soon though. Let me convert this to a Roadmap item and we will get this sorted for you!

**Reply by mvarendorff** [reply]
This is certainly a usecase for us. We use drafts + preview for writers to, well, preview their articles before publishing.

The current state now leads to a case where a writer can partially fill out a block and then the graphql request fails because a required field is null, causing a  blank page in the application. At least in that regard, I am not sure I would consider this just a feature request if a state is allowed that blows up the graphql endpoint.

As a sidenote, I am open to looking into making a PR for this, once it's determined what a solution to this could look like.

**Comment by silveltman** [comment]
I understand what you mean and actually see the benefit in this. For me a solution could be the following:

- using drafts for incomplete documents (not shown in visual preview)
- using publish for complete documents (shown in visual preview)
- and a 3rd option which I would use to push to my production website (shown in visual preview, but not on production website)

Names for these buttons could be: **Save draft**, **Publish**, and **Build site**
However, I would like to be able to change the label in the buttons, so I can make Publish > Complete for example.
This way I could have a global 'Publish' button (or Build Site) which sends a webhook to build the production branche.

**Comment by abernh** [comment]
This was added via the collection config `drafts.validate = true` -- see https://payloadcms.com/docs/versions/drafts#options
